3days/data/src/main/java/com/weave/data/extension/HandleApiResponse.kt

```
@@ -0,0 +1,17 @@
+import com.weave.model.network.NetworkError
+import com.weave.model.network.NetworkResult
+
+suspend fun <T> handleApiResponse(apiCall: suspend () -> retrofit2.Response<T>): NetworkResult<T> {
+    return try {
+        val response = apiCall()
+        if (response.isSuccessful) {
+            NetworkResult.Success(response.body()!!)
+        } else {
+            NetworkResult.Error(NetworkError.getNetworkErrorByCode(response.code()))
+        }
+    } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)